CHANGES.md

```diff
@@ -0,0 +1,12 @@
+<!--
+Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+
+SPDX-License-Identifier: curl
+-->
+
+In a release tarball, check the RELEASES-NOTES file for what was done in the
+most recent release. In a git check-out, that file mentions changes that have
+been done since the previous release.
+
+See the online [changelog](https://curl.se/changes.html) for the edited and
+human readable version of what has changed in different curl releases.
```

Comment:
Wouldn't it be useful to add the link above also to `RELEASE-NOTES`?

Reply:
Yeah probably. I fix.

Reply:
Sorry, what I meant to suggest was to add the <https://curl.se/changes.html> link into the `RELEASE-NOTES` text.

It's a useful link, and e.g. in curl-for-win I stop distributing `CHANGES.md`, so the link will not be exposed, but in `RELEASE-NOTES` it would be!

(This update is also good regardless)

Reply:
Ah ok. Yes, we can add a link to the RELEASE-NOTES, but I think I'll do that in a separate PR since I also need to make sure some other scripts handle that proper.

Reply:
The GHA job for checking links did not like the markdown link to a local file anyway so I will revert that.